pkg/config/msp.go

```diff
@@ -589,3 +615,56 @@ func addRootCAToOrgMSP(org *cb.ConfigGroup, rootCA *x509.Certificate) error {
 
 	return nil
 }
+
+func getOrgMSPValue(org *cb.ConfigGroup) (*cb.ConfigValue, error) {
+	configValue, ok := org.Values[MSPKey]
+	if !ok {
+		return nil, errors.New("org doesn't have MSP value")
+	}
+	return configValue, nil
+}
+
+// RevokeCertificateFromMSP takes a variadic list of x509 certificates, creates
+// a new CRL signed by the specified ca certificate and private key, and appends
+// it to the revocation list for the specified application org MSP.
+func RevokeCertificateFromMSP(config *cb.Config, orgName string, caCert *x509.Certificate, caPrivKey *ecdsa.PrivateKey, certs ...*x509.Certificate) error {
+	org, err := getApplicationOrg(config, orgName)
+	if err != nil {
+		return err
+	}
+
+	fabricMSPConfig, err := getFabricMSPConfig(org)
+	if err != nil {
+		return fmt.Errorf("getting msp config: %v", err)
+	}
+
+	// TODO validate that this certificate was issued by this MSP
+
+	revokeTime := time.Now()
+	revokedCertificates := make([]pkix.RevokedCertificate, len(certs))
+	for i, cert := range certs {
+		revokedCertificates[i] = pkix.RevokedCertificate{
+			SerialNumber:   cert.SerialNumber,
+			RevocationTime: revokeTime,
+		}
+	}
+
+	crlBytes, err := caCert.CreateCRL(rand.Reader, caPrivKey, revokedCertificates, revokeTime, revokeTime.Add(365*24*time.Hour))
```

Comment:
I don't usually say this but you may want to pull the 365*24*time.Hour out to a constant just above the function. 

IIUC, most CRL's in the real world expire pretty much every day... Not something I expect for fabric but I'm not really sure what the recommendation would be either.